## **IMPORTANDO OS DADOS**

In [27]:
from dbfread import DBF
import pandas as pd
import os
diretorio_dbf = r'C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB'

# Lista para armazenar DataFrames
dataframes = []

# Percorre todos os arquivos no diretório
for arquivo in os.listdir(diretorio_dbf):
    if arquivo.endswith('.dbf'):
        caminho_completo = os.path.join(diretorio_dbf, arquivo)
        
        # Lê o arquivo .dbf
        try:
            dbf = DBF(caminho_completo)
            df = pd.DataFrame(iter(dbf))  # Converte para DataFrame
            dataframes.append(df)  # Adiciona o DataFrame à lista
            print(f'Lido: {caminho_completo}')
        except Exception as e:
            print(f'Erro ao ler {caminho_completo}: {e}')

# Concatena todos os DataFrames em um único
if dataframes:
    df_total = pd.concat(dataframes, ignore_index=True)
    print('Todos os dados concatenados com sucesso!')
else:
    print('Nenhum arquivo .dbf encontrado ou lido.')

# Agora df_total contém todos os dados
#

Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2000.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2001.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2002.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2003.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2004.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\PB\DNPB2005.dbf
Lido: C:\Users\PICHAU\Desktop\menu\Estudos\Estudos UFPB\Tec. Pesquisa e Análise de Dados\Análise DATASUS SINASC\Ignorar\tratados\P

In [25]:
display(df_total.columns)

Index(['contador', 'LOCNASC', 'CODMUNNASC', 'IDADEMAE', 'ESTCIVMAE', 'ESCMAE',
       'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES', 'GESTACAO', 'GRAVIDEZ',
       'PARTO', 'CONSULTAS', 'DTNASC', 'SEXO', 'APGAR1', 'APGAR5', 'RACACOR',
       'PESO', 'CODANOMAL', 'CODESTAB', 'CODOCUPMAE', 'IDANOMAL', 'CODBAINASC',
       'CODBAIRES', 'UFINFORM', 'HORANASC', 'DTCADASTRO', 'DTRECEBIM',
       'ORIGEM', 'CODCART', 'NUMREGCART', 'DTREGCART', 'CODPAISRES',
       'NUMEROLOTE', 'VERSAOSIST', 'DIFDATA', 'DTRECORIG', 'NATURALMAE',
       'CODMUNNATU', 'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT',
       'QTDPARTNOR', 'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC',
       'TPMETESTIM', 'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART',
       'STCESPARTO', 'TPROBSON', 'STDNEPIDEM', 'STDNNOVA', 'RACACOR_RN',
       'RACACORN', 'ESCMAE2010', 'CODMUNCART', 'CODUFNATU', 'TPNASCASSI',
       'ESCMAEAGR1', 'DTRECORIGA', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC',
       'PARIDADE', 'KOTELCHU

## **ETL**

In [4]:
dataframes

[      contador LOCNASC CODMUNNASC IDADEMAE ESTCIVMAE ESCMAE QTDFILVIVO  \
 0            1       1    1100049       39         2      3         06   
 1            2       1    1100122       18         1      3         00   
 2            3       1    1100155       19         2      5         00   
 3            4       1    1502202       19         5      2         02   
 4            5       2    1503705       16         2      3              
 ...        ...     ...        ...      ...       ...    ...        ...   
 57422    57423       1    5201108       37         1      9         00   
 57423    57424       1    5201108       31         4      5         00   
 57424    57425       1    5201108       29         1      9         02   
 57425    57426       1    5203807       19         5      3         02   
 57426    57427       1    5300108       31         2      1         03   
 
       QTDFILMORT CODMUNRES GESTACAO GRAVIDEZ PARTO CONSULTAS    DTNASC SEXO  \
 0             00 

In [5]:
df_total['DTNASC'] = pd.to_datetime(df_total['DTNASC'], format='%d%m%Y')
df_total['DTNASC']  
#comentario

0         2000-09-17
1         2000-07-23
2         2000-07-17
3         2000-05-26
4         2000-05-26
             ...    
1412694   2023-12-20
1412695   2023-07-17
1412696   2023-07-11
1412697   2023-10-14
1412698   2023-12-07
Name: DTNASC, Length: 1412699, dtype: datetime64[ns]

analisando os valores NaN's

In [6]:
df_nans = df_total['PESO'].loc[pd.isna(df_total['PESO'])]

df_nans

Series([], Name: PESO, dtype: object)

In [7]:
# Substituir vírgulas por pontos e remover espaços
df_total['PESO'] = df_total['PESO'].str.replace(',', '.').str.strip()

# Converter para float, forçando valores inválidos a serem NaN
df_total['PESO'] = pd.to_numeric(df_total['PESO'], errors='coerce')

# Agora você pode filtrar normalmente os bebês com peso menor ou igual a 2500
df_baixo_peso = df_total.loc[(df_total['PESO'] < 2500) & (df_total['PESO'] > 0)].reset_index()

print(df_baixo_peso['PESO'].sort_values())

53553     100.0
53554     100.0
95406     100.0
93543     100.0
53058     110.0
          ...  
88342    2498.0
97242    2498.0
97248    2498.0
71758    2498.0
96926    2498.0
Name: PESO, Length: 99908, dtype: float64


Então nosso dado amostral será de 99908 casos onde foi registrado o nascimento abaixo do peso

In [8]:
df_baixo_peso

index contador LOCNASC CODMUNNASC IDADEMAE ESTCIVMAE ESCMAE  \
0            5        6       2    1503705       21         2      4   
1           17       18       1    2202208       36         2      2   
2           18       19       1    2202208       36         2      2   
3           20       21       1    2202208       20         3      3   
4           46       47       1    2211001       33         2      3   
...        ...      ...     ...        ...      ...       ...    ...   
99903  1412651      NaN       1     261160       29         2      5   
99904  1412654      NaN       1     261350       31         2      4   
99905  1412658      NaN       1     261390       28         2      4   
99906  1412659      NaN       1     261390       32         2      5   
99907  1412673      NaN       1     261390       27         2      4   

      QTDFILVIVO QTDFILMORT CODMUNRES  ... CODUFNATU TPNASCASSI ESCMAEAGR1  \
0             01         01   2503704  ...       NaN        NaN        NaN   
1             12         00   2502508  ...       NaN        NaN        NaN   
2             11         00   2502508  ...       NaN        NaN        NaN   
3                        01   2502508  ...       NaN        NaN        NaN   
4             01         00   2503100  ...       NaN        NaN        NaN   
...          ...        ...       ...  ...       ...        ...        ...   
99903         00         00    250400  ...        25          1         08   
99904         01         01    251335  ...        25          1         12   
99905         00         00    251230  ...        26          1         05   
99906         01         00    250900  ...        25          1         08   
99907         01         00    250900  ...        26          1         06   

      DTRECORIGA TPFUNCRESP TPDOCRESP DTDECLARAC PARIDADE KOTELCHUCK  CONTADOR  
0            NaN        NaN       NaN        NaN      NaN        NaN       NaN  
1            NaN        NaN       NaN        NaN      NaN        NaN       NaN  
2            NaN        NaN       NaN        NaN      NaN        NaN       NaN  
3            NaN        NaN       NaN        NaN      NaN        NaN       NaN  
4            NaN        NaN       NaN        NaN      NaN        NaN       NaN  
...          ...        ...       ...        ...      ...        ...       ...  
99903                     2         3   20122023        0          5   1683232  
99904                     2         3   23072023        1          5   1686902  
99905                               3   18072023        0          5   1687696  
99906                               3   31072023        1          5   1687723  
99907                               3   31102023        1          5   1688303  

[99908 rows x 73 columns]

**A partir de agora chamaremos a variável *df_baixo_peso* para chamar os dados dos nascimentos abaixo do peso.**

# **Iniciando coleta de informações - PARAIBA**

Municipios da paraiba com maiores indices, e menores indices

In [9]:
# Filtrar os municípios cujo código começa com "25"
df_filtrado = df_baixo_peso.loc[df_baixo_peso['CODMUNNASC'].astype(str).str.startswith('25')]
media_peso_municipio = df_filtrado.groupby('CODMUNNASC')['PESO'].count().sort_values()

print(media_peso_municipio.tail(10))


CODMUNNASC
2510808     1677
251370      1705
251620      1946
250370      2054
250630      2106
2504009     3829
251080      5947
2507507     9745
250400     22323
250750     31013
Name: PESO, dtype: int64


ANALISE DOS MUNICIPIOS DE RESIDENCIA

In [10]:
# Filtrar os municípios cujo código começa com "25"
df_res = df_baixo_peso.loc[df_baixo_peso['CODMUNRES'].astype(str).str.startswith('25')]
media_peso_municipio_res = df_res.groupby('CODMUNRES')['PESO'].count().sort_values()

print(media_peso_municipio_res.tail(10))


CODMUNRES
250320      1311
250370      1318
251620      1404
250180      1777
2504009     2008
251080      2017
251370      3049
2507507     5465
250400      9142
250750     16689
Name: PESO, dtype: int64


Media da idade das mães que dão luz a bebês abaixo do peso:

In [11]:
df_baixo_peso['IDADEMAE'] = pd.to_numeric(df_baixo_peso['IDADEMAE'], errors='coerce')
print(df_baixo_peso['IDADEMAE'].mean())
print(df_baixo_peso['IDADEMAE'].median())

25.564478153272752
25.0


Sexo do recém nascido

In [12]:
# Filtrar os municípios cujo código começa com "25"
df_sexo = df_baixo_peso
# Agrupar pelos códigos filtrados e contar os registros de peso, ordenando por contagem
sexo = df_sexo.groupby('SEXO')['PESO'].count().sort_values()
# Exibir as médias
print(sexo)

SEXO
9        7
I        8
0      169
M     1976
F     2132
1    44679
2    50937
Name: PESO, dtype: int64


Tipo de Gravidez (Tipo de gravidez: 1– Única; 2– Dupla; 3– Tripla ou mais; 9–
Ignorado.)

In [13]:
# Filtrar os municípios cujo código começa com "25"
df_grav = df_baixo_peso
# Agrupar pelos códigos filtrados e contar os registros de peso, ordenando por contagem
tipo_gravidez = df_grav.groupby('GRAVIDEZ')['PESO'].count().sort_values()
# Exibir as médias
print(tipo_gravidez)

GRAVIDEZ
9       82
       121
3      530
2    13281
1    85894
Name: PESO, dtype: int64


Análise pré-natal (Número de consultas de pré‐natal. Valores: 1– Nenhuma; 2– de
1 a 3; 3– de 4 a 6; 4– 7 e mais; 9– Ignorado. )

In [14]:

df_nat = df_baixo_peso
pre_natal = df_nat.groupby('CONSULTAS')['PESO'].count().sort_values()
print(pre_natal)

CONSULTAS
8        1
       276
9     2061
1     4832
2    11398
3    40049
4    41291
Name: PESO, dtype: int64


Tipo de Parto (Tipo de parto: 1– Vaginal; 2– Cesário; 9– Ignorado )

In [15]:

df_part = df_baixo_peso
contagem_parto = df_part.groupby('PARTO')['PESO'].count().sort_values()
print(contagem_parto)

PARTO
9       37
       109
2    49464
1    50298
Name: PESO, dtype: int64


Apresenta Anomalia Congênita (Código da anomalia (CID 10) )

In [16]:

df_part = df_baixo_peso

anomalia = df_part.groupby('CODANOMAL')['PESO'].count().sort_values()

print(anomalia.tail(10))

CODANOMAL
Q02X       57
Q664       57
Q909       75
Q549       80
Q668       96
Q793      102
Q898      103
Q000      150
Q899      231
        97313
Name: PESO, dtype: int64


Duração de gestação (Semanas de gestação: 1– Menos de 22 semanas; 2– 22 a 27
semanas; 3– 28 a 31 semanas; 4– 32 a 36 semanas; 5– 37 a 41
semanas; 6– 42 semanas e mais; 9– Ignorado. )

In [17]:

df_part = df_baixo_peso

gestacao = df_part.groupby('GESTACAO')['PESO'].count().sort_values()

print(gestacao.tail(10))

GESTACAO
1      261
9      517
6      727
      1314
2     4299
3     8749
5    40799
4    43242
Name: PESO, dtype: int64


Recorrencia de idades das mães

In [18]:

df_mae = df_baixo_peso

mae = df_mae.groupby('IDADEMAE')['PESO'].count().sort_values()
print(mae.tail(10))

IDADEMAE
26.0    4312
25.0    4658
24.0    4691
23.0    4871
17.0    4901
22.0    5175
21.0    5266
18.0    5284
19.0    5432
20.0    5520
Name: PESO, dtype: int64


taxa de nascimentos de bebês abaixo do peso Visão paraíba:

In [19]:
soma_nascidos_abaixo_peso = df_baixo_peso['index'].count()
soma_total_nascidos = df_total['LOCNASC'].count()
taxa_total_pb = (soma_nascidos_abaixo_peso/soma_total_nascidos)*100
# df_total
# 1412699 
print(f'{taxa_total_pb:.2f}%')


7.07%


taxa de nascimentos de bebês abaixo do peso Visão municipal:

In [20]:
# Filtrar os municípios cujo código começa com "25"
# df_res = df_baixo_peso.loc[df_baixo_peso['CODMUNRES'].astype(str).str.startswith('25')]
# media_peso_municipio_res = df_res.groupby('CODMUNRES')['PESO'].count().sort_values()

# Filtrar os municípios cujo código começa com "25"
df_res = df_baixo_peso.loc[df_baixo_peso['CODMUNRES'].astype(str).str.startswith('25')]
df_total_res = df_total.loc[df_total['CODMUNRES'].astype(str).str.startswith('25')]
nascidos_abaixo_peso_por_municipio = df_res.groupby('CODMUNRES')['PESO'].count()
total_nascidos_por_municipio = df_total_res.groupby('CODMUNRES')['PESO'].count()
taxa_por_municipio = (nascidos_abaixo_peso_por_municipio / total_nascidos_por_municipio) * 100

print(taxa_por_municipio.sort_values(ascending=False).head(10))


CODMUNRES
250000     11.764706
2501401    10.163551
250115      9.715243
2505279     9.618875
2512606     9.583333
250205      9.562564
251300      9.234508
2508604     9.163347
2512762     9.061489
2508554     9.055118
Name: PESO, dtype: float64


Divisão por região de nascimento

In [21]:
df_total['ESCMAEAGR1']

0          NaN
1          NaN
2          NaN
3          NaN
4          NaN
          ... 
1412694     08
1412695     08
1412696     06
1412697     06
1412698     06
Name: ESCMAEAGR1, Length: 1412699, dtype: object